src/main/java/com/squareup/kotlinpoet/Parameter.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.kotlinpoet
+
+import java.lang.reflect.Type
+import kotlin.reflect.KClass
+
+data class Parameter(val name: String? = null, val type: TypeName) {
```

Comment:
This competes with an existing class, `ParameterSpec`. I’m anxious about having both types that mean almost the exact same thing. Would it be simpler to just use `ParameterSpec` in `LambdaTypeName` ?

Reply:
That's a good point. We'll need only name and type, should we just ignore any other params set on the `ParameterSpec`?

Reply:
Hmm, the problem is that `name` is mandatory for `ParameterSpec`, and optional for `LambdaTypeName`. Will switching from `name: String` to `name: String?` on `ParameterSpec` be justified for that purpose? Then we could have `ParameterSpec.ofType()` to create an unnamed parameter.

Reply:
That sounds a bit weird.

Reply:
How about we permit an empty string as the name in those cases, and require a factory method to get instances like that:

```
  ParameterSpec string = ParameterSpec.unnamed(String::class)
```

Reply:
I'll give it a try